src/cargo/util/semver_ext.rs

```diff
@@ -217,18 +225,18 @@ mod matches_prerelease {
             //
             (">1.2.3, <1.2.4", "1.2.3-0", false),
             (">1.2.3, <1.2.4", "1.2.3-1", false),
-            (">1.2.3, <1.2.4", "1.2.4-0", false), // upper bound semantic
+            (">1.2.3, <1.2.4", "1.2.4-0", true), // upper bound semantic
```

Comment:
I've not followed all of the conversation but this change seems odd.

Technically, its also an odd test case as I would expect this to be an empty range

Reply:
If we are going to follow the upper bound semantic in the RFC, this change seems wrong. The equivalent requirements will be `>1.2.3, <1.2.4-0`, which doesn't include 1.2.4. The comment highlights them not because they need to be flipped, but something we must discuss and decide.

Reply:
Oh no, I missed this piece in the [Unresolved questions](https://rust-lang.github.io/rfcs/3493-precise-pre-release-cargo-update.html#unresolved-questions). 

The RFC  suggests to pad '0' for `Less op`,  considering all partial version included, the semantic will change the `Less op`
to
```
<I.J.K equivalent to < I.J.K-0
<I.J equivalent to <I.J.0-0
<I equivalent to <I.0.0-0
```

And in another way, if we always treat the "<I.J.K" as "<I.J.K-0", then peple will imposible to speficy that any prerelease version less than `I.J.K`, that seems a problem also?

Reply:
Hence the conundrum.  One way to solve is messy: only determine how to evaluate `<` bounds by looking at any lower bounds that also exist.

Reply:
> only determine how to evaluate < bounds by looking at any lower bounds that also exist.

I had updated the semantic for `Less Op` now it will meet the `upper bound semantic`

Reply:
> only determine how to evaluate < bounds by looking at any lower bounds that also exist.

I have updated the `Less Op` semantic. Now it will pad prerelase with "0"